### ISSUE TITLE
Local execution with "run" not respecting filter

### ISSUE THREAD

**Issue by bdice** [glotzerlab/signac-flow#83]
### Description

I want to locally execute an operation for a subset of jobs, but it executes on all jobs instead.

### init.py:
```
import signac

project = signac.init_project('execution_filter_issue')

for a in range(10):
    for b in range(10):
        project.open_job({'a': a, 'b': b}).init()
```

### project.py:
```
from flow import FlowProject

@FlowProject.operation
def print_job(job):
    print(job, job.sp)

if __name__ == '__main__':
    FlowProject().main()
```

Then
1. `python project.py run -f a 0` runs all jobs, rather than just those where `job.sp.a == 0`. _(unexpected)_
2. `python project.py run -f a 0 b 0` runs only one job, as expected.
3. `python project.py run -f a 0 b` runs all jobs. _(unexpected)_

### Error output

The filter interpretation is correct: `Interpreted filter arguments as '{"a": 0}'.`

### System configuration

Please complete the following information:

 - Operating System [e.g. macOS]: macOS and Linux
 - Version of Python [e.g. 3.7]: 3.6.7
 - Version of signac [e.g. 1.0]: 1.0.0
 - Version of signac-flow: 0.6.4

**Comment by csadorf** [comment]
This issue only occurs in the 0.7 line, not in a release version. However it is a serious bug. I'll fix it.

**Comment by csadorf** [comment]
Fixed with merge of PR #84 .
